models/articles.model.js

```diff
@@ -0,0 +1,19 @@
+const db = require("../db/connection");
+
+
+async function selectArticleById(id) {
+    try {
+        const article = await db.query(`
+        SELECT articles.author, articles.title, articles.article_id, articles.body, articles.topic, articles.created_at, SUM(comments.votes)::INTEGER AS votes, articles.article_img_url FROM articles JOIN comments ON articles.article_id = comments.article_id WHERE articles.article_id=$1
+        GROUP BY articles.article_id;`, [id]);
+        if (article.rows.length === 0) {
+            return Promise.reject({status:404, message:'article not found'})
+        }
+        // some logic to alter the date can go here?
+        return article.rows[0];
+    } catch(error) {
```

Comment:
This catch block in the model is unnecessary because we are already handling errors in the controller where this promise chain is continued and response/err is handled (note the then block after we invoke our model in the controller).